cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,553 @@
+        agg_data["lastused"] = agg_data["last_seen"]
+        result.append(agg_data)
+
+    logger.info(
+        f"Aggregated {len(role_assumptions)} events into {len(result)} unique role assumption relationships"
+    )
+    return result
+
+
+def _extract_role_assumption_from_event(
+    event: Dict[str, Any],
+    region: str,
+    current_aws_account_id: str,
+) -> Optional[Dict[str, Any]]:
+    """
+    Extract role assumption details from a single CloudTrail event.
+
+    :type event: Dict[str, Any]
+    :param event: Single CloudTrail event from lookup_events API
+    :type region: str
+    :param region: The AWS region where the event occurred
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :rtype: Optional[Dict[str, Any]]
+    :return: Role assumption record or None if event is not a role assumption
+    """
+    event_name = event.get("EventName")
+
+    # Only process STS role assumption events
+    if event_name not in [
+        "AssumeRole",
+        "AssumeRoleWithSAML",
+        "AssumeRoleWithWebIdentity",
+    ]:
+        return None
+
+    # Extract source principal (who is assuming the role)
+    source_principal = _extract_source_principal(event)
+    if not source_principal:
+        logger.debug(
+            f"Could not extract source principal from event {event.get('EventId')}"
+        )
+        return None
+
+    # Extract destination principal (role being assumed)
+    destination_principal = _extract_destination_principal(event)
+    if not destination_principal:
+        logger.debug(
+            f"Could not extract destination principal from event {event.get('EventId')}"
+        )
+        return None
+
+    # Build the standardized role assumption record
+    assumption = {
+        "SourcePrincipal": source_principal,
+        "DestinationPrincipal": destination_principal,
+        "Action": event_name,
+        "EventId": event.get("EventId"),
+        "EventTime": event.get("EventTime"),
+        "SourceIPAddress": event.get("SourceIPAddress"),
+        "UserAgent": event.get("UserAgent"),
+        "AwsRegion": event.get("AwsRegion", region),
+        "AccountId": current_aws_account_id,
+        "AssumedRoleArn": destination_principal,  # For relationship targeting
+        "PrincipalArn": source_principal,  # For relationship targeting
+    }
+
+    # Add additional context from CloudTrail event JSON
+    cloudtrail_event = _parse_cloudtrail_event_json(event.get("CloudTrailEvent"))
+    if cloudtrail_event:
+        assumption.update(
+            {
+                "SessionName": _extract_session_name(cloudtrail_event, event_name),
+                "RequestId": cloudtrail_event.get("requestID"),
+                "RecipientAccountId": cloudtrail_event.get("recipientAccountId"),
+            }
+        )
+
+    return assumption
+
+
+def _extract_source_principal(event: Dict[str, Any]) -> Optional[str]:
```

Comment:
Same point about this looking overly complicated. Can we not get away with just direct dict accesses?